### PR TITLE
feat(frontend): Card component for Token Group

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

We want to create a custom Token Card but for groups of tokens. Basically, this card will collect a group of tokens and once clicked, it will expand to show them. While it is collapsed, it will have the same aspects of the other Token Cards.

# Note

Before being used, we MUST create a listener for all the tokens in the group. We cannot create it inside the `IF` condition, otherwise the summary will not be updated (since the `Listener` will not be mounted)

# Changes

- Create type for LogoSize, so that we can pass it as parameter and bubble it up.
- Add prop `logoSize` in components `TokenLogo`, `TokenCard` and `TokenCardContent`.
- Change `TokenCard` to accept a number of tokens: this prop will be used if the `TokenCard` is acting as a summary card for the group.
- Create component TokenGroupCard that has:
  - A main "summary" card that re-uses the normal `TokenCard`
  - Some sub-`TokenCard` that are built/shown when the main one is clicked
- We pass a custom created "summary" token to the "summary" token card, that has all the properties of the Native Token of the group plus a few changed properties for appearance and the financial data summirized.

# Tests

I mocked a token group, uniting ICP with BTC and giving fake data.

https://github.com/user-attachments/assets/0235bef3-cd11-44f0-97c2-162e71f57191

https://github.com/user-attachments/assets/c8e4ce34-4b2b-41ea-a8ef-da5c1f96a60b

### CODE REVIEW

**Review by darioloetsc — APPROVED** [review]

**Comment by AntonioVentilii-DFINITY** [comment]
[OISY-69]

File: src/frontend/src/lib/components/tokens/TokenCard.svelte [hunk 1]
```diff
@@ -1,21 +1,31 @@
...
 	import TokenName from '$lib/components/tokens/TokenName.svelte';
 	import TokenSymbol from '$lib/components/tokens/TokenSymbol.svelte';
 	import Card from '$lib/components/ui/Card.svelte';
 	import { TOKEN_CARD, TOKEN_GROUP } from '$lib/constants/test-ids.constants';
+	import type { LogoSize } from '$lib/types/components';
 	import type { CardData } from '$lib/types/token-card';
 
 	export let data: CardData;
 	export let testIdPrefix: typeof TOKEN_CARD | typeof TOKEN_GROUP = TOKEN_CARD;
+	export let tokenCount: number | undefined = undefined;
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/tokens/TokenCard.svelte:13]: Should `tokenCount` actually becomes an optional variable or `CardData`?

File: src/frontend/src/lib/components/tokens/TokenCard.svelte [hunk 2]
```diff
@@ -1,21 +1,31 @@
...
 
 <Card noMargin testId={`${testIdPrefix}-${data.symbol}`}>
 	<TokenSymbol {data} />
 
 	<TokenName {data} slot="description" />
 
-	<TokenLogo {data} slot="icon" color="white" />
+	<TokenLogo
+		{data}
+		subLogo={nonNullish(tokenCount) ? { type: 'tokenCount', count: tokenCount } : undefined}
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/tokens/TokenCard.svelte:24]: Unrelated to this PR: just realized that `subLogo` could potentially be called `badge`.

File: src/frontend/src/lib/components/tokens/TokenGroupCard.svelte [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	import TokenCardWithOnClick from '$lib/components/tokens/TokenCardWithOnClick.svelte';
+	import TokenCardWithUrl from '$lib/components/tokens/TokenCardWithUrl.svelte';
+	import { TOKEN_GROUP } from '$lib/constants/test-ids.constants';
+	import { SLIDE_PARAMS } from '$lib/constants/transition.constants';
+	import type { TokenUiGroup } from '$lib/types/token';
+	import type { CardData } from '$lib/types/token-card';
+
+	export let tokenGroup: TokenUiGroup;
+
+	let isOpened = false;
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/tokens/TokenGroupCard.svelte:14]: `isExpanded` for consistency? I noticed that term used in another PR.

File: src/frontend/src/lib/components/tokens/TokenGroupCard.svelte [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+	let { nativeToken } = tokenGroup;
+
+	let headerData: CardData;
+	$: headerData = {
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/tokens/TokenGroupCard.svelte:19]: this can go in a util in another PR. like `mapGroupToCardData`

File: src/frontend/src/lib/components/tokens/TokenCard.svelte (hunk 2 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/tokens/TokenCard.svelte:24]: good one!

- PR #3084

File: src/frontend/src/lib/components/tokens/TokenGroupCard.svelte (hunk 3 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/tokens/TokenGroupCard.svelte:14]: yes, makes sense, it is kind of a `dropdown`

File: src/frontend/src/lib/components/tokens/TokenGroupCard.svelte (hunk 4 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/tokens/TokenGroupCard.svelte]: Sure, done:

- PR #3086

File: src/frontend/src/lib/components/tokens/TokenCard.svelte (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/tokens/TokenCard.svelte]: yes, makes sense, moved

**Comment by AntonioVentilii-DFINITY** [comment]
@peterpeterparker ready for another review

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx
